tiledb/sm/cpp_api/array_schema_experimental.h

```diff
@@ -156,6 +157,38 @@ class ArraySchemaExperimental {
     return DimensionLabel(ctx, dim_label);
   }
 
+  /**
+   * Returns a copy of the schema's array currentDomain. To change the
+   * currentDomain, use `set_current_domain()`.
+   *
+   * @param ctx The TileDB context.
+   * @param array_schema The array schema.
+   * @return Copy of the schema's currentDomain.
+   */
+  static CurrentDomain current_domain(
+      const Context& ctx, const ArraySchema& array_schema) {
+    tiledb_current_domain_t* current_domain;
+    ctx.handle_error(tiledb_array_schema_get_current_domain(
+        ctx.ptr().get(), array_schema.ptr().get(), &current_domain));
+
+    return CurrentDomain(ctx, current_domain);
+  }
+
+  /**
+   * Sets the currentDomain.
+   *
+   * @param ctx The TileDB context.
+   * @param array_schema The array schema.
+   * @param current_domain The currentDomain to use.
+   */
+  static void set_current_domain(
+      const Context& ctx,
+      const ArraySchema& array_schema,
+      const CurrentDomain& current_domain) {
+    ctx.handle_error(tiledb_array_schema_set_current_domain(
+        ctx.ptr().get(), array_schema.ptr().get(), current_domain.ptr().get()));
+  }
```

Comment:
These functions don't have to accept a context, it is being used only for the last error, and they can get the context from the schema.

Reply:
These are static methods, this is why ctx is required.

Reply:
Hmm, other experimental methods already explicitly accept a context. OK then. 🤷🏻‍♂️